jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/AsyncContent.java

```diff
@@ -51,8 +51,8 @@ public String toString()
 
     private final AutoLock.WithCondition lock = new AutoLock.WithCondition();
     private final SerializedInvoker invoker = new SerializedInvoker();
-    private final Queue<AsyncChunk> chunks = new ArrayDeque<>();
-    private Content.Chunk.Error errorChunk;
+    private final Queue<Content.Chunk> chunks = new ArrayDeque<>();
```

Comment:
This seems just to complicate things, adding more `instanceof`.
Is it really necessary? I'd revert to `AsyncChunk`.

Reply:
AsyncChunk cannot carry a non-persistent Failure (aka a warning).  So it is necessary.